src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrState = [];
+  let newState = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === 'clear') {
+      newState = {};
+      arrState.push({ ...newState });
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((key) => delete newState[key]);
+      arrState.push({ ...newState });
+    }
+
+    if (action.type === 'addProperties') {
+      newState = {
+        ...newState,
+        ...action.extraData,
+      };
+      arrState.push({ ...newState });
```

Comment:
[CODE STYLE] - Again, to adhere to the [DONT REPEAT YOURSELF] principle, remove the push operation from inside the condition and ensure it is done once after all cases have been processed.